### PR TITLE
Functional tests - Add new test 'FO > User account > Get GDPR data in csv'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | Add new test 'FO > User account > Get GDPR data in csv'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | no
| Possible impacts? | no

### CODE REVIEW

File: tests/UI/campaigns/functional/FO/03_userAccount/05_getGDPRDataInCSV.js [hunk 1]
```diff
@@ -0,0 +1,968 @@
...
+        await expect(ipAddress).to.not.be.null;
+      });
+    });
+
+    describe('Check GDPR data in CSV', async () => {
+      it('should check general info', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', 'checkGeneralInfo', baseContext);
+
+        // To replace isVisible after release of the module psgdpr
+        // const age = await basicHelper.age(customerData.birthDate);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/03_userAccount/05_getGDPRDataInCSV.js:236]:
```suggestion
        // TODO
        // To replace isVisible after release of the module psgdpr
        // const age = await basicHelper.age(customerData.birthDate);
```

Adding maybe a `TODO` to not forget about this

File: tests/UI/utils/files.js [hunk 2]
```diff
@@ -138,8 +138,8 @@ module.exports = {
    * @param ignoreTimeZone {boolean} True to delete timezone string added to some image url
```
Comment (Progi1984) [inline tests/UI/utils/files.js:138]:
```suggestion
   * @param ignoreTimeZone {boolean} True to delete timezone string added to some image url
   * @param encoding {string} Encoding for the file
```

File: tests/UI/campaigns/functional/FO/03_userAccount/05_getGDPRDataInCSV.js [hunk 3]
```diff
@@ -0,0 +1,968 @@
...
+  firstName: 'Marc',
+  lastName: 'Beier',
+  country: 'France',
+  address: '17, Main street',
+  city: 'Paris',
+  company: 'PrestaShop',
+});
+
+const contactUsData = new ContactUsFakerData(
+  {
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/05_getGDPRDataInCSV.js:73]:
```suggestion
const contactUsData = new ContactUsFakerData({
```

**Review by Progi1984 — COMMENTED** [review]
Small feedback

File: tests/UI/utils/files.js (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/utils/files.js:138]: Done thx :+1:

**Review by Progi1984 — APPROVED** [review]

File: tests/UI/campaigns/functional/FO/03_userAccount/05_getGDPRDataInCSV.js [hunk 4]
```diff
@@ -0,0 +1,971 @@
...
+        );
+        await expect(isVisible, 'The data in Last connections table is not correct!').to.be.true;
+      });
+
+      it('should check that Newsletter subscription table is empty', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', 'checkNewsletterSubscriptionTable', baseContext);
+
+        const isVisible = await files.isTextInFile(
+          filePath,
+          '"MODULE:NEWSLETTERSUBSCRIPTION""Newslettersubscription:noemailtoexport,thiscustomerhasnotregistered.""',
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/03_userAccount/05_getGDPRDataInCSV.js:324]: Maybe we can create customizable functions (on a helper) to get what to check.

**Review by Progi1984 — CHANGES_REQUESTED** [review]
@nesrineabdmouleh Could you rebase your PR ? You have some conflicting files.

**Review by Progi1984 — APPROVED** [review]

**Review by SD1982 — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Pull request merged**